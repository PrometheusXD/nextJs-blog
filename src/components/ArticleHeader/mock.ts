import { ArticleHeaderProps } from '.';

export default {
  id: '603e56b939bd976b42d790a0',
  title: 'Vis nobis veritus maluisset id, nonumy adversarium',
  excerpt:
    'Quod oblique mei ne, cetero delicatissimi sea cu, cu est harum dicit verterem. Tota forensibus ut usu, luptatum dignissim persecuti vel ex. Mea nostrud instructior cu, et referrentur theophrastus eum. Pro in iuvaret senserit.',
  cover: {
    id: '603e5a2785540677309fd51b',
    alternativeText: '',
    url: 'https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg',
  },
  author: {
    id: '603e557239bd976b42d79094',
    displayName: 'Ot√°vio Miranda',
    slug: 'otavio-miranda',
  },
  categories: [
    {
      id: '603e55b539bd976b42d79097',
      displayName: 'JavaScript',
      slug: 'java-script',
    },
    {
      id: '603e55ef39bd976b42d7909a',
      displayName: 'React',
      slug: 'react',
    },
  ],
  createdAt: '2020-04-20',
} as ArticleHeaderProps;
