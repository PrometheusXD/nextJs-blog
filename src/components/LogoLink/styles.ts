import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  color: inherit;

  > img {
    width: 14rem;
    height: 14rem;
    border-radius: 50%;
  }
`;
