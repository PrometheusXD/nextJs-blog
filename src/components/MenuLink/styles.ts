import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display:block;
    padding-bottom: ${theme.spacings.small};
    font-size: 1.8;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 300ms ease-in-out;

    :hover {
      border-right: 0.5rem solid ${theme.colors.secondary};
    }

    > a {
      color:${theme.colors.white};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      :hover {
        color: ${theme.colors.secondary};
      }
    }

  `}
`;
