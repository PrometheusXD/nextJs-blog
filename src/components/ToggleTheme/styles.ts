import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
`;

export const Label = styled.label`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    font-size: 0;
    overflow: hidden;
    color: rgba(0,0,0,0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
      border: 1px solid ${theme.colors.secondary};

      &:hover {
        border: 1px solid ${theme.colors.darkerGray};
      }
    }

    &:checked + ${Slider}:before {
      transform: translateX(1.9rem);
      background: ${theme.colors.primary};
    }

    &:hover + ${Slider} {
      border: 1px solid ${theme.colors.secondary};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkerGray};
    border: 1px solid ${theme.colors.darkerGray};
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
    border-radius: 2rem;

    &:before {
      position: absolute;
      content: "";
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.12rem;
      background: ${theme.colors.white};
      -webkit-transition: all 300ms ease-in-out;
      transition: all 300ms ease-in-out;
      border-radius: 50%;

    }
  `}
`;
