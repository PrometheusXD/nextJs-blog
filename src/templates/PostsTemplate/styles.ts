import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;
    transition: all 300ms ease-in-out;


    &:disabled,
    &:disabled:hover {
      background: ${theme.colors.darkerGray};
      color: #FFFFFF;
    }

    &:hover {
      background: ${theme.colors.secondary};
      color: #FFFFFF;
    }
  `}
`;
