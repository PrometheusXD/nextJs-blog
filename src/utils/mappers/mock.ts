export default {
  data: {
    setting: {
      data: {
        id: '1',
        attributes: {
          blogName: 'Matheus Rodrigues',
          blogDescription:
            'ncdsbnci csnb cus cu ybsduc bsdcb isb icbs bcsc lsnc sklcnl nkncl s.',
          menuLink: [
            {
              id: '1',
              link: '#bla',
              text: 'bla',
              newTab: false,
            },
          ],
          logo: {
            data: {
              id: '3',
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/dcfobv3yq/image/upload/v1686858611/logo_a47d56dba6.svg',
              },
            },
          },
          text: 'cedcedcedcedcedce',
        },
      },
    },
    posts: {
      data: [
        {
          id: '2',
          attributes: {
            slug: 'creating-an-api-with-strapi',
            title: 'creating an api with strapi',
            excerpt:
              'creating a fucking api with the shit of strapi who everyone love but i hate',
            content:
              'lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum ',
            createdAt: '2023-06-10T21:11:24.338Z',
            allowComments: false,
            cover: {
              data: null,
            },
            categories: {
              data: [
                {
                  id: '3',
                  attributes: {
                    displayName: 'Python',
                    slug: 'python',
                  },
                },
              ],
            },
            tags: {
              data: [
                {
                  id: '1',
                  attributes: {
                    displayName: 'useState',
                    slug: 'use-state',
                  },
                },
              ],
            },
            author: {
              data: {
                id: '2',
                attributes: {
                  displayName: 'lucas da silva',
                  slug: 'lucas-da-silva',
                },
              },
            },
          },
        },
        {
          id: '1',
          attributes: {
            slug: 'posnvdjknvj-jcnscn-sinc-ions-ncsn-jcnsmcmko',
            title: 'posnvdjknvj jcnscn sinc ions ncsn jcnsmcmko ',
            excerpt:
              'cndsicvdsiivcnsdui cnids ncisdb cisbic bsdib cisdbci bsicb sjkcb jksnc nscnsk klkwmcons jocnjias ncjsnc ljsnj cnsjnc sd jlcjsndioic nsdc sjkdcnsonc sn clsnc jsdjvbjajoisn kj njkn jsb jsb.\njcncjksjkcnsk.',
            content: '# jciodncodncod',
            createdAt: '2023-06-09T11:46:12.511Z',
            allowComments: true,
            cover: {
              data: null,
            },
            categories: {
              data: [
                {
                  id: '2',
                  attributes: {
                    displayName: 'React',
                    slug: 'react',
                  },
                },
                {
                  id: '1',
                  attributes: {
                    displayName: 'JavaScript',
                    slug: 'java-script',
                  },
                },
              ],
            },
            tags: {
              data: [
                {
                  id: '2',
                  attributes: {
                    displayName: 'useEffect',
                    slug: 'use-effect',
                  },
                },
                {
                  id: '1',
                  attributes: {
                    displayName: 'useState',
                    slug: 'use-state',
                  },
                },
              ],
            },
            author: {
              data: {
                id: '1',
                attributes: {
                  displayName: 'Matheus Rodrigues',
                  slug: 'matheus-rodrigues',
                },
              },
            },
          },
        },
      ],
    },
  },
};
