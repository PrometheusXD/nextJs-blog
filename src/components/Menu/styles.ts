import styled, { DefaultTheme, css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

const wrapperChanger = (theme: DefaultTheme) =>
  css`
    left: -30rem;
    overflow-y: hidden;

    @media ${theme.media.lteMedium} {
    left: -32rem;
    }
`;

export const Wrapper = styled.div<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;

    ${!menuVisible && wrapperChanger(theme)}
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

const buttonChanger = (theme: DefaultTheme) =>
  css`
    left: 1rem;

    @media ${theme.media.lteMedium} {
    left: -0.5rem;
    }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;

    &:hover {
    color: ${theme.colors.secondary};
    }

    ${!menuVisible && buttonChanger(theme)}
  `}
`;
