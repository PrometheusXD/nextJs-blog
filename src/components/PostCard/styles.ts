import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    box-shadow: 0 0 5px 0px ${theme.colors.darkerGray};

    &::after,
    &::before {
      content: "";
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      transition: transform .3s ease-in-out;
    }

    &::before {
      border-left: 2px solid ${theme.colors.primary};
      border-right: 2px solid ${theme.colors.primary};
      transform: scaleY(0);
    }

    &::after {
      border-top: 2px solid ${theme.colors.primary};
      border-bottom: 2px solid ${theme.colors.primary};
      transform: scaleX(0);
    }

    &:hover::before {
      transform: scaleY(1);
    }

    &:hover::after {
      transform: scaleX(1);
    }


    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    padding: calc(${theme.spacings.small} - 0.6rem);
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.darkText};
  `}
`;
