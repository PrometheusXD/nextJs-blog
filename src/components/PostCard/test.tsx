import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';

import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render a card with heading, cover and excerpt', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.cover.url,
    );
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();

    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      `/post/${props.slug}`,
    );

    expect(container).toMatchSnapshot();
  });
});
