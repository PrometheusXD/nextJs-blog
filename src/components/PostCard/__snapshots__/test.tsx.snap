// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<PostCard /> should render a card with heading, cover and excerpt 1`
] = `
.c4 {
  color: #030518;
  line-height: 1.2;
  font-size: 2.4rem;
}

.c0 {
  position: relative;
}

.c0::after,
.c0::before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c0::before {
  border-left: 2.5px solid #f231a5;
  border-right: 2.5px solid #f231a5;
  -webkit-transform: scaleY(0);
  -ms-transform: scaleY(0);
  transform: scaleY(0);
}

.c0::after {
  border-top: 2.5px solid #f231a5;
  border-bottom: 2.5px solid #f231a5;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
}

.c0:hover::before {
  -webkit-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.c0:hover::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c0:hover .c3 {
  color: #f231a5;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 .c3 {
  margin: 0;
  margin-bottom: 1.6rem;
}

.c1 {
  max-width: 100%;
}

.c2 {
  padding: calc(1.6rem - 0.6rem);
}

.c5 {
  color: #333333;
}

<div>
  <a
    href="/post/vis-nobis-veritus-maluisset-id-nonumy-adversarium"
  >
    <div
      class="c0"
    >
      <img
        alt="Vis nobis veritus maluisset id, nonumy adversarium"
        class="c1"
        src="https://res.cloudinary.com/dlizakp2a/image/upload/v1614698965/Ubuntu_gel_by_Midge_Mantissa_Sinnaeve_7caea7ddb4.jpg"
      />
      <div
        class="c2"
      >
        <h2
          class="c3 c4"
        >
          Vis nobis veritus maluisset id, nonumy adversarium
        </h2>
        <p
          class="c5"
        >
          Fugit aliquip erroribus eu eam, an mel vidit doctus. Vis nobis veritus maluisset id, nonumy adversarium neglegentur est ut. Tation persius pri eu, modo minim fastidii mei no. Ad tota pertinax definiebas eam.
        </p>
      </div>
    </div>
  </a>
</div>
`;
